.github/workflows/maven.yml

```diff
@@ -24,3 +24,30 @@ jobs:
         cache: maven
     - name: Build with Maven
       run: mvn -B test --file pom.xml
+    - name: Generate JaCoCo Badge
+      id: jacoco
+      uses: cicirello/jacoco-badge-generator@v2
+      with:
+          generate-branches-badge: true
+          generate-summary: true
+          fail-if-coverage-less-than: 50
+          fail-on-coverage-decrease: true
+
+    - name: Log coverage percentage(if it changed)
+      if: ${{ github.event_name == 'push' }}
```

Comment:
This is only true when the event name is "push" what about when it's a pull request?

Reply:
The thinking behind this is that no permanent coverage will be created for pull requests. This plugin creates images that are commited into our git repo and makes sense when pushing new code into main where an updated image will be created and added in a new commit. 
When a pull request triggers this action file no updated badge will be created, only the step before will run and fail the check if code coverage decreases.

Reply:
Ahh ok.